packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -159,6 +168,8 @@ export class SharedTreeCore<
 			new Set(this.summaryElements.map((e) => e.key)).size === this.summaryElements.length,
 			0x350 /* Index summary element keys must be unique */,
 		);
+
+		this.editor = this.changeFamily.buildEditor((change) => this.applyChange(change), anchors);
```

Comment:
Currently all editors use `ProgressiveEditBuilderBase`, which is stateful and was designed to accumulate the edits for a transaction. Since this PR adds explicit support for tracking the edits in a transaction we should investigate whether we can remove `ProgressiveEditBuilder` and `ProgressiveEditBuilderBase`.

Reply:
That's a good point. It's weird how it is currently, because each checkout/branch owns a builder, which is accumulating changes for the whole life of the branch. We don't read the changes, but if we did, we'd see all the changes that got squashed when that checkout committed its transactions.

Reply:
I made a work item to track replacing the old transaction style with the new one, and made a note in there to try removing ProgressiveEditBuilder at that time.